app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt

```diff
@@ -579,39 +602,123 @@ open class UserViewModel(
+                  val oldRequestIds = oldRequests.map { it.id }.toSet()
+                  val newRequests = requests.filterNot { oldRequestIds.contains(it.id) }
+
+                  if (newRequests.isNotEmpty()) {
+                    val newRequest = newRequests.first()
+                    val senderUser = users.find { it.id == newRequest.from }
+                    val senderName = senderUser?.name ?: context?.getString(R.string.unknown)
+
+                    // Existing notification for a new friend request
+                    if (context != null) {
+                      NotificationHelper.showNotification(
+                          context = context,
+                          notificationId = 1001,
+                          title = context.getString(R.string.new_friend_request),
+                          text = context.getString(R.string.friend_request_message, senderName),
+                          iconResId = R.drawable.app_logo,
+                          priority = NotificationCompat.PRIORITY_HIGH)
+                    }
+                  }
+                }
+              }
             },
             onFailure = { exception ->
               Log.e("USER_VIEW_MODEL", "Failed to listen to friend requests: ${exception.message}")
             })
   }
+
+  /**
+   * Sets up a listener for sent friend requests so that the UI updates in real-time. This method
+   * also triggers a notification if a sent friend request is accepted.
+   */
+  fun getSentFriendRequests() {
+    val userId = Firebase.auth.uid.orEmpty()
+    if (userId.isEmpty()) return
+
+    val oldSentRequests = _sentFriendRequests.value
+    sentFriendRequestsListener?.remove()
+    sentFriendRequestsListener =
+        friendRequestRepository.listenToSentFriendRequests(
+            userId = userId,
+            onSuccess = { requests ->
+              val oldRequestsMap = oldSentRequests.associateBy { it.id }
+              _sentFriendRequests.value = requests
+
+              // Find requests that have just transitioned to accepted = true
+              val newlyAccepted =
+                  requests.filter { newReq ->
+                    newReq.accepted && oldRequestsMap[newReq.id]?.accepted == false
+                  }
+
+              if (newlyAccepted.isNotEmpty()) {
+                newlyAccepted.forEach { acceptedReq ->
+                  // The "to" user is the one who accepted
+                  getUsersByIds(listOf(acceptedReq.to)) { users ->
+                    val acceptorUser = users.firstOrNull()
+                    val acceptorName = acceptorUser?.name ?: context?.getString(R.string.unknown)
+
+                    // Show the notification that the request was accepted
+                    context?.let {
+                      NotificationHelper.showNotification(
+                          context = context,
+                          notificationId = 1002,
+                          title = it.getString(R.string.friend_request_accepted),
```

Comment:
Here 1002 is used as a hardcoded value. Maybe create a constant and explain it's value. 1002 Just seems random

Reply:
Actually it is quite random, any unique id can be set in this field. I added a comment to the code to explain this.

Reply:
From what I read online, the ID should still be different from notfication to notification, so using the current timestamp and creating a (pseudo) random ID out of it might be the stolution. Take [this](https://stackoverflow.com/questions/12978184/android-get-unique-id-of-notification) as an example, it is in java but should be easily converted to kotlin.

Reply:
As if you use the same id twice, it just updates the notification, it doesn't create a new one (which would be the desired behaviour)

Reply:
When I though about this I thought that we should only keep the latest notifications such that the user is not getting too many notifications, but your suggestion seems to be actually better, thank you, I will implement it as soon as possible.

Reply:
Done!